django/django_contracts/admin/contract.py

```diff
@@ -0,0 +1,93 @@
+from typing import List, Optional, Tuple, Union
+
+from django.contrib import admin
+from django.db.models import QuerySet
+from django.http import HttpRequest
+
+from django_contracts.models import (
+    LegalContract,
+    LegalContractVersion,
+    PublishableMixin,
+)
+from django_contracts.models.publishable import PublishStatus
+
+
+def publish(modeladmin, request, queryset: QuerySet[PublishableMixin]) -> None:
+    for obj in queryset:
+        obj.publish()
+
+
+publish.short_description = "Publish"
+
+
+@admin.register(LegalContract)
+class LegalContractAdmin(admin.ModelAdmin):
+    actions = (publish,)
+    readonly_fields = (
+        "datetime_published",
+        "publish_status",
+        "latest",
+    )
+
+    list_display = (
+        "__str__",
+        "publish_status",
+        "datetime_published",
+    )
+    list_filter = ("publish_status",)
+
+    def get_readonly_fields(
+        self, request: HttpRequest, obj: Optional[LegalContract] = None
+    ) -> Union[List[str], Tuple]:
+        if not obj or obj.publish_status != PublishStatus.PUBLISHED:
+            return self.readonly_fields
+        return self.readonly_fields + ("slug",)
+
+    def has_delete_permission(
+        self, request: HttpRequest, obj: Optional[PublishableMixin] = None
+    ) -> bool:
+        if obj and obj.publish_status != PublishStatus.DRAFT:
+            return False
+        return super().has_delete_permission(request, obj)
+
+
+@admin.register(LegalContractVersion)
+class LegalContractVersionAdmin(admin.ModelAdmin):
+    actions = (publish,)
```

Comment:
Since new versions will probably have only minor differences to old ones, having a "copy as new version" action might be nice?

Reply:
Was considering that as well but figured I didn't want to add more code since I don't see the copying happening that often. The reason a publish action was added here is because that's about the only way to add actions in Django admin (i.e. can't add buttons or such on the detail page easily), and I wanted to make that state management happen through code.